### PR TITLE
feat:[#4] Add statuses changes to providers and catalogues

### LINKED ISSUE #4: Assign Statuses to Providers and Update Statuses for Services and Catalogs
Implement status functionality for providers and improve status management for services and catalogs. Each item will have associated actions (buttons) that trigger specific statuses:

CREATE: UNPUBLISHED
PUBLISH: PUBLISHED
UNPUBLISH: UNPUBLISHED
SUSPEND: SUSPENDED
DELETE: DELETED

Remove all unnecessary actions (for eg. publish us unverified)

Documentation: https://docs.cyfronet.pl/display/FID/Konkretne+zadania

### PR DESCRIPTION
**Pull request by goreck888**
- Add services to manage statuses
- Add controllers to call services
   
Closes #4

### CODE REVIEW

**Comment by Marcelinna** [comment]
when the Provider is deleted, button Suspended, Publish, Unpublish should be no visible
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

1. Delete Catalogue
2. Click on the deleted catalogue
image
4. See problem
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

1. Unpublish provider in Backoffice https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/backoffice/providers/eosc.100percentit
2. Bo to Markeplace https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/providers/eosc.100percentit
3. See problem - Provider is visible

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Button Delete doesn't work
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
 Change name from draft to unpublished
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
Change name from Stop showing in Marketplace to Unpublish
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
In Filter we have Unpublish I Unpublished, Delete unpublish
image

**Comment by Marcelinna** [comment]
Why when I filter by Unpublished, I see only one service, which I unpublished a while ago, I should see all services witch unpublished status
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
I can remove catalogue with connected services
I don't see message "This Catalogue has services connected to it, therefore is not possible to remove it."

File: app/policies/api/v1/ess/bundle_policy.rb [hunk 1]
```diff
@@ -3,11 +3,7 @@
 class Api::V1::Ess::BundlePolicy < Api::V1::EssPolicy
   class Scope < Scope
     def resolve
-      scope.joins(:service).where(
-        "bundles.status = ? AND services.status IN (?)",
-        "published",
-        %w[published unverified]
-      )
+      scope.joins(:service).where("bundles.status = ? AND services.status IN (?)", "published", %w[published])
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/policies/api/v1/ess/bundle_policy.rb:6]: Done.

File: app/controllers/backoffice/catalogues_controller.rb [hunk 2]
```diff
@@ -24,11 +24,11 @@ def edit
 
   def destroy
     if Catalogue::Delete.call(@catalogue.id)
-      redirect_to backoffice_catalogues_path, notice: "Catalogue removed successfully"
+      flash[:notice] = "Catalogue removed successfully"
     else
-      redirect_to backoffice_catalogues_path,
-                  alert: "This Catalogue has services connected to it, therefore is not possible to remove it."
+      flash[:alert] = "This Catalogue has services connected to it, therefore is not possible to remove it."
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/controllers/backoffice/catalogues_controller.rb:29]: For now, it is the only reason, but I agree, in future development, it should be more generic.

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**